tests/test_routes.py

```diff
@@ -141,6 +141,337 @@ def test_cves_returns_422_for_non_existing_package_status(self):
+        payloads.cve3["codename"] = "new_name3"
+        payloads.cve4["codename"] = "new_name4"
+
+        update_cves_response1 = self.client.put(
+            "/security/cves.json",
+            json=[payloads.cve2],
+        )
+
+        update_cves_response2 = self.client.put(
+            "/security/cves.json",
+            json=[payloads.cve3],
+        )
+
+        update_cves_response3 = self.client.put(
+            "/security/cves.json",
+            json=[payloads.cve4],
+        )
+
+        assert update_cves_response1.status_code == 200
+        assert update_cves_response2.status_code == 200
+        assert update_cves_response3.status_code == 200
+
+        # Check sorting by updated_at field (desc)
+        sorted_cves2_desc = self.client.get(
+            "/security/cves.json?sort_by=updated"
+        ).json
+
+        assert (
+            sorted_cves2_desc["cves"][0]["updated_at"]
+            > sorted_cves2_desc["cves"][1]["updated_at"]
+        )
+        assert (
+            sorted_cves2_desc["cves"][1]["updated_at"]
+            > sorted_cves2_desc["cves"][2]["updated_at"]
+        )
+
+        # Check ordering by asc value
+        sorted_cves2_asc = self.client.get(
+            "/security/cves.json?sort_by=updated&order=ascending"
+        ).json
+
+        assert (
+            sorted_cves2_asc["cves"][0]["updated_at"]
+            < sorted_cves2_asc["cves"][1]["updated_at"]
+        )
+        assert (
+            sorted_cves2_asc["cves"][1]["updated_at"]
+            < sorted_cves2_asc["cves"][2]["updated_at"]
+        )
+
+    def test_bulk_upsert_cves_returns_422_for_invalid_cve(self):
+        cve = payloads.cve1.copy()
+        cve["hello"] = "world"
+
+        response = self.client.put("/security/cves.json", json=[cve])
+
+        assert response.status_code == 422
+        assert "Unknown field." in response.json["errors"]
+
+    def test_bulk_upsert_cves(self):
+        response_1 = self.client.get(
+            f"/security/cves/{self.models['cve'].id}.json"
+        )
+        assert response_1.status_code == 200
+
+        response_2 = self.client.get(
+            f"/security/cves/{payloads.cve1['id']}.json"
+        )
+        assert response_2.status_code == 404
+
+        response_3 = self.client.put(
+            "/security/cves.json",
+            json=[
+                payloads.cve1,
+                payloads.cve2,
+            ],
+        )
+        assert response_3.status_code == 200
+
+        response = self.client.get(
+            f"/security/cves/{payloads.cve1['id']}.json"
+        )
+        assert response.status_code == 200
+
+        response = self.client.get(
+            f"/security/cves/{payloads.cve2['id']}.json"
+        )
+        assert response.status_code == 200
+
+    def test_delete_non_existing_cve_returns_404(self):
+        response = self.client.delete(
+            f"/security/cves/{payloads.cve1['id']}.json"
+        )
+
+        assert response.status_code == 404
```

Comment:
I'm not quite sure `404` is the most informative status code for this case, maybe a `400: Bad Request` would be more appropriate? But I recognise that you didn't actually write this test, it only appears in the diff because you moved it, so it's out of scope for this PR. Definitely don't fix it here, just wondering if it's worth filing an issue about it? What do you think?

Reply:
Ignore me here. I'm not sure about this at all. There's interesting discussion in https://stackoverflow.com/questions/6439416/status-code-when-deleting-a-resource-using-http-delete-for-the-second-time, but `404` is as good an answer to this question as any probably.